### PR TITLE
adds std and var refs and var prim

### PR DESCRIPTION
**Pull request by ngimel**
Per title

### CODE REVIEW

**Review by mruberry — APPROVED** [review]
Awesome!

**Comment by ngimel** [comment]
@ezyang I'm getting test errors with `no _refs support for prims.var.default` after rebase, do you know how to fix that? https://github.com/pytorch/pytorch/runs/6535300685?check_suite_focus=true

**Comment by ezyang** [comment]
It's because you forgot to add these refs to`__all__`

File: torch/_prims/__init__.py [hunk 1]
```diff
@@ -1828,10 +1829,22 @@ def _bool_return_reduction_meta(inp, dims):
...
+    else:
+        output_dtype = inp.dtype
+    return _reduction_meta(inp, dims, output_dtype=output_dtype)
+
+
 _sum_doc = """
     Computes the sum of elements in the input tensor over the list of dimensions
     specified in the dim argument
     """
+_prod_doc = """
```
Comment (jjsjann123) [inline torch/_prims/__init__.py:1844]: @ngimel We forgot to update the `_prod_doc` in `prod = _make_reduction_prim` :wink:

**Pull request closed without merging**